plugin/src/main/scala/org/ehcache/integrations/play/EhcacheJCacheWrapperModule.scala

```diff
@@ -63,11 +83,34 @@ class EhcacheValueWrapper extends ValueWrapper {
   */
 class EhcacheJCacheWrapper(xmlConfig: Option[XmlConfiguration]) extends JCacheWrapper {
 
+  var enhanced = Set.empty[String]
+
   def valueWrapper(name: String) = {
-    new EhcacheValueWrapper
+    enhanced contains name match {
+      case true => new EhcacheValueWrapper
+      case false => new NoOpValueWrapper
+    }
   }
 
   def enhanceConfiguration(name: String, baseConfig: JCacheConfiguration[String, Any]): JCacheConfiguration[String, Any] = {
-    baseConfig
+    xmlConfig match {
+      case Some(config) =>
+        val template: CacheConfigurationBuilder[String, Any] = config
+                .newCacheConfigurationBuilderFromTemplate(name, classOf[String], classOf[Any])
+        template match {
+          case null => generateMinimalConfiguration(name)
+          case t if t.hasConfiguredExpiry => fromEhcacheCacheConfiguration(t)
+          case t =>
+            enhanced = enhanced + name
+            fromEhcacheCacheConfiguration(t.withExpiry(new WrappedValueWithExpiryExpiration))
+        }
+      case None => generateMinimalConfiguration(name)
+    }
+  }
+
+  def generateMinimalConfiguration(name: String): JCacheConfiguration[String, Any] = {
+    enhanced = enhanced + name
+    fromEhcacheCacheConfiguration(newCacheConfigurationBuilder(classOf[String], classOf[Any], heap(Int.MaxValue))
```

Comment:
Ehcache 3 does not really have a way to create a config based on a JCache config. And given we own both ways, I would not bother right now.